File: ai-services/ai-orchestrator/app/models/generated_project.py
from pydantic import BaseModel, Field

from app.models.project_limits import (
    MAX_FILE_CONTENT_LENGTH,
    MAX_FILE_PATH_LENGTH,
    MAX_FILE_TYPE_LENGTH,
    MAX_PROJECT_FILES,
    MAX_PROJECT_NAME_LENGTH,
)
from app.models.project_type import ProjectType


class GeneratedFile(BaseModel):
    file_path: str = Field(alias="filePath", min_length=1, max_length=MAX_FILE_PATH_LENGTH)
    file_type: str = Field(alias="fileType", min_length=1, max_length=MAX_FILE_TYPE_LENGTH)
    content: str = Field(max_length=MAX_FILE_CONTENT_LENGTH)


class GeneratedProject(BaseModel):
    project_name: str = Field(alias="projectName", min_length=1, max_length=MAX_PROJECT_NAME_LENGTH)
    project_type: ProjectType = Field(default="html", alias="projectType")
    files: list[GeneratedFile] = Field(min_length=1, max_length=MAX_PROJECT_FILES)

File: frontend/src/types/generatedProject.ts
export type ProjectType = 'html' | 'vue' | 'react'

export type DeploymentTarget = 'docker' | 'github-actions' | 'kubernetes'

export type GeneratedFile = {
  filePath: string
  fileType: string
  content: string
}

export type GeneratedProject = {
  projectName: string
  projectType?: ProjectType
  files: GeneratedFile[]
}

export type AppSummary = {
  id: string
  userId: string
  appName: string
  description: string
  type: string
  status: string
  deployUrl: string | null
  createTime: string
}

export type AppVersion = {
  id: string
  appId: string
  versionNo: number
  prompt: string
  project: GeneratedProject
  createTime: string
}

export type GenerationResult = {
  app: AppSummary
  version: AppVersion
  project: GeneratedProject
}

export type Deployment = {
  id: string
  appId: string
  versionNo: number
  projectType: ProjectType
  artifactUrl: string
  target: DeploymentTarget
  status: 'planned' | 'running' | 'succeeded' | 'failed' | 'skipped'
  plannedCommands: string[]
  executionLogs: string[]
  accessUrl: string | null
  createTime: string | null
}
